test/reduce/structured_loop_to_selection_reduction_pass_test.cpp

```diff
@@ -0,0 +1,2372 @@
+               OpMemberDecorate %4 0 Offset 0
+               OpMemberDecorate %4 1 Offset 4
+               OpMemberDecorate %4 2 Offset 8
+               OpMemberDecorate %4 3 Offset 12
+               OpDecorate %4 Block
+               OpDecorate %5 DescriptorSet 0
+               OpDecorate %5 Binding 0
+               OpDecorate %3 Location 0
+          %6 = OpTypeVoid
+          %7 = OpTypeFunction %6
+          %8 = OpTypeBool
+         %10 = OpTypeInt 32 1
+          %4 = OpTypeStruct %10 %10 %10 %10
+         %11 = OpTypePointer Uniform %4
+          %5 = OpVariable %11 Uniform
+         %12 = OpConstant %10 0
+         %13 = OpTypePointer Uniform %10
+         %14 = OpTypeInt 32 0
+         %15 = OpConstant %14 0
+         %16 = OpConstant %10 1
+         %17 = OpConstant %10 2
+         %18 = OpConstant %10 3
+         %20 = OpConstantFalse %8
+         %21 = OpTypeFloat 32
+         %22 = OpTypeVector %21 4
+         %23 = OpTypePointer Output %22
+          %3 = OpVariable %23 Output
+        %114 = OpUndef %10
+        %115 = OpUndef %8
+        %116 = OpConstantTrue %8
+          %2 = OpFunction %6 None %7
+         %24 = OpLabel
+         %31 = OpAccessChain %13 %5 %12
+         %32 = OpLoad %10 %31
+         %33 = OpINotEqual %8 %32 %15
+         %34 = OpAccessChain %13 %5 %16
+         %35 = OpLoad %10 %34
+         %36 = OpINotEqual %8 %35 %15
+         %37 = OpAccessChain %13 %5 %17
+         %38 = OpLoad %10 %37
+         %39 = OpINotEqual %8 %38 %15
+         %40 = OpAccessChain %13 %5 %18
+         %41 = OpLoad %10 %40
+         %42 = OpINotEqual %8 %41 %15
+               OpBranch %44
+         %44 = OpLabel
+         %98 = OpPhi %10 %12 %24 %114 %46
+         %97 = OpPhi %8 %33 %24 %115 %46
+               OpSelectionMerge %45 None	; Was OpLoopMerge %45 %46 None
+               OpBranchConditional %97 %49 %45
+         %49 = OpLabel
+               OpSelectionMerge %51 None
+               OpBranchConditional %33 %52 %51
+         %52 = OpLabel
+         %55 = OpIAdd %10 %98 %16
+               OpBranch %51
+         %51 = OpLabel
+        %100 = OpPhi %10 %98 %49 %55 %52
+        %113 = OpSelect %8 %33 %36 %97
+               OpSelectionMerge %57 None
+               OpBranchConditional %36 %58 %57
+         %58 = OpLabel
+         %60 = OpIAdd %10 %100 %16
+         %63 = OpLogicalOr %8 %113 %33
+               OpSelectionMerge %65 None
+               OpBranchConditional %39 %66 %65
+         %66 = OpLabel
+         %68 = OpIAdd %10 %100 %18
+         %70 = OpLogicalNot %8 %63
+               OpBranch %65 	; Was OpBranch %46
+         %65 = OpLabel
+         %72 = OpLogicalOr %8 %63 %20
+               OpBranch %57     ; Was OpBranch %46
+         %57 = OpLabel
+               OpBranch %73
+         %73 = OpLabel
+         %99 = OpPhi %10 %100 %57 %114 %75
+               OpSelectionMerge %74 None ; Was OpLoopMerge %74 %75 None
+               OpBranchConditional %116 %76 %74
+         %76 = OpLabel
+  
```

Comment:
What happens if OpLoad were another OpAccessChain feeding an OpLoad?

Reply:
I've added a test for that case, and made this work in a storage class-aware way.